### PR TITLE
Make main menubar of lepton-schematic wholly in Scheme.

### PR DESCRIPTION
**Pull request by vzh**
- Get rid of obsolete limited functions in `s_menu.c` and the file itself.
- Redefine main menubar using Scheme FFI; get rid of scheme-in-c code in `x_menus.c`.

### CODE REVIEW

File: libleptongui/scheme/schematic/ffi/gobject.scm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+;;; This program is distributed in the hope that it will be useful,
+;;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;;; GNU General Public License for more details.
+;;;
+;;; You should have received a copy of the GNU General Public License
+;;; along with this program; if not, write to the Free Software
+;;; Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+(define-module (schematic ffi gtk)
```
Comment (graahnul-grom) [inline libleptongui/scheme/schematic/ffi/gobject.scm:19]: `(schematic ffi gobject)`?

Comment (vzh) [reply libleptongui/scheme/schematic/ffi/gobject.scm]: Of course. My fault.

File: libleptongui/src/x_menus.c [hunk 2]
```diff
@@ -306,33 +146,18 @@ get_main_popup (GschemToplevel* w_current)
...
-    if (e.stock_id != NULL && gtk_stock_lookup (e.stock_id, &stock_info)) {
-      gtk_action_set_stock_id (GTK_ACTION (action), e.stock_id);
-    } else {
-      gtk_action_set_icon_name (GTK_ACTION (action), e.stock_id);
-    }
-
-    /* Connect things up so that the actions get run */
-    g_signal_connect (G_OBJECT (action), "activate",
-                      G_CALLBACK (g_menu_execute), w_current);
+    action = make_menu_action (e.action,
```
Comment (graahnul-grom) [inline libleptongui/src/x_menus.c:156]: `unused variable 'stock_info' [-Wunused-variable]` in `get_main_popup()`

File: libleptongui/scheme/schematic/ffi/gobject.scm [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+;;; You should have received a copy of the GNU General Public License
+;;; along with this program; if not, write to the Free Software
+;;; Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+(define-module (schematic ffi gobject)
+  #:use-module (system foreign)
+
+  #:export (g_object_set_data))
+
+(define libgobject (dynamic-link "libgobject-2.0"))
```
Comment (graahnul-grom) [inline libleptongui/scheme/schematic/ffi/gobject.scm:24]: Let us not forget to look up later how this library is named in `Cygwin`.

**Review by graahnul-grom — APPROVED** [review]
Everything works as expected, AFAICT.

File: libleptongui/scheme/schematic/ffi/gobject.scm (hunk 3 above)
Comment (vzh) [reply libleptongui/scheme/schematic/ffi/gobject.scm:24]: Good catch. I saw `cyggobject-2.0-0.dll` somewhere.  Could you please check?

File: libleptongui/src/x_menus.c (hunk 2 above)
Comment (vzh) [reply libleptongui/src/x_menus.c:156]: Need to fix this, too.

File: libleptongui/scheme/schematic/ffi/gobject.scm (hunk 3 above)
Comment (graahnul-grom) [reply libleptongui/scheme/schematic/ffi/gobject.scm:24]: Yes, that is the name, I've just checked in VirtualBox.

**Comment by vzh** [comment]
@graahnul-grom, being concerned with the issue about `gobject` in Cygwin. WDYT, should we first fix it before merging the branch?

**Comment by graahnul-grom** [comment]
I think, yes. As experience shows, checking with `Cygwin` can help
us discover not-so-evident bugs. Please take a look at [these changes](https://github.com/graahnul-grom/lepton-eda/commits/gui-make-main-menubar-wholly-in-scheme).

**Comment by vzh** [comment]
Thank you!

**Comment by graahnul-grom** [comment]
I'm sorry, I forgot to modify the `nobase_dist_scmdata_DATA`,
`EXTRA_DIST` and `CLEANFILES` lists in `Makefile.am`.
Thank you for fixing this!

**Comment by graahnul-grom** [comment]
@vzh: `nobase_scmdata_DATA` should have been adjusted, too.
My bad. :-(
Should I open a PR, or could it be committed directly to `master`?

```diff
diff --git a/libleptongui/scheme/Makefile.am b/libleptongui/scheme/Makefile.am
index 66f39fa6d..c36ee4b50 100644
--- a/libleptongui/scheme/Makefile.am
+++ b/libleptongui/scheme/Makefile.am
@@ -45,6 +45,7 @@ nobase_dist_scmdata_DATA = \
 
 nobase_scmdata_DATA = \
 	schematic/gschemdoc.scm \
+	schematic/ffi/gobject.scm \
 	schematic/ffi/gtk.scm \
 	schematic/ffi.scm
 
```

[GOBJECT_SCM.diff.txt](https://github.com/lepton-eda/lepton-eda/files/5754089/GOBJECT_SCM.diff.txt)

**Comment by vzh** [comment]
Committed, thanks.

**Pull request merged**